### ISSUE TITLE
Warn user if cluster did not come up as expected

### ISSUE THREAD

**Issue by rmetzger** [stratosphere/stratosphere#266]
While I did some work on a cluster, I was wondering why my job did not utilize all TaskManagers.
It seems that I started my job too early (before all TaskManager registered with the JobManager) and therefore, the compiler did not consider them.

We should either make the `start-cluster.sh` script blocking (with a timeout). Or the pact-client.sh should report a warning if less TaskManagers than expected (number in `slaves` file) are up.

**Comment by fhueske** [comment]
I opt for make the call blocking.
Users that don't want a blocking call can run:

```
./start-cluster.sh &
```

The pact-client may not run on the same machine as the JM and have no access to its slave file. So that would not work in the general case.

I have some code somewhere that is doing a blocking call until all TM register (until some timeout). Would be a quick fix.
Any objections?

**Comment by StephanEwen** [comment]
I like the idea. Do you want to start the TMs asynchronously and then wait
until the JM Log says that all are registered? What about people who
decrease the Log Level?

**Comment by fhueske** [comment]
Yes, start all TMs (as it is done right now) and wait until each TM generates a log message. 
That would be an easy solution. I guess, we can check the log-level in the log4j config file.

**Comment by StephanEwen** [comment]
I think waiting for the TM logs works only in the NFS setup, otherwise they
are on different machines.

**Comment by rmetzger** [comment]
Honestly, I think this feature is a nice-to-have and we should not put too much energy into it right now. (other systems like Hadoop also don't check if the cluster came up properly) 
(the scripts are utilities, I guess professional setups have their own monitoring tools (hue, ambari, cloudera manager)

I suggest to wait until the JobManager REST Interface ( https://github.com/stratosphere/stratosphere/issues/297 ) is ready and than query if for the number of JobManagers using a simple `curl` request.

**Comment by fhueske** [comment]
I meant the JM logs about connected TMs in the JM logs.
